### PR TITLE
Handle HTML entity in mailto

### PR DESCRIPTION
**Pull request by shawniverson**
Fix https://github.com/MailScanner/v5/issues/335

### CODE REVIEW

File: common/usr/share/MailScanner/perl/MailScanner/Message.pm [hunk 1]
```diff
@@ -7683,9 +7689,14 @@ sub DisarmEndtagCallback {
         # https://github.com/MailScanner/v5/issues/229
         MailScanner::Log::DebugLog("DisarmLinkURL = $DisarmLinkURL");
         MailScanner::Log::DebugLog("linkurl = $linkurl");
-        if ($DisarmLinkURL =~ m/^ma[il]+to[:;]/i && $linkurl =~ /@/ ) {
-           my @list = split(/@/, $linkurl);
-           $linkurl = $list[1];
+        if ($DisarmLinkURL =~ m/^ma[il]+to[:;]/i ) {
```
Comment (Skywalker-11) [inline common/usr/share/MailScanner/perl/MailScanner/Message.pm]: What use case does this have `[il]+` ? `malito:`??

File: common/usr/share/MailScanner/perl/MailScanner/Message.pm [hunk 2]
```diff
@@ -7795,12 +7806,17 @@ sub DisarmEndtagCallback {
         MailScanner::Log::DebugLog("DisarmLinkURL = $DisarmLinkURL");
         MailScanner::Log::DebugLog("linkurl = $linkurl");
 
-        if ($DisarmLinkURL =~ m/^ma[il]+to[:;]/i && $linkurl =~ /@/ ) {
-           my @list = split(/@/, $linkurl);
-           if ( $emailuser ne "" && $list[0] ne $emailuser) {
-             $alarm = 1;
-           }
-           $linkurl = $list[1];
+        if ($DisarmLinkURL =~ m/^ma[il]+to[:;]/i ) {
```
Comment (Skywalker-11) [inline common/usr/share/MailScanner/perl/MailScanner/Message.pm]: Same here

File: common/usr/share/MailScanner/perl/MailScanner/Message.pm [hunk 3]
```diff
@@ -7573,16 +7574,21 @@ sub DisarmEndtagCallback {
...
-       # Remove any leading or trailing text
-       $squashedtext =~ s/^.*\s+(?=.*\@)//;
-       $squashedtext =~ s/\s+.*$//;
-       # Remove &gt; and &lt; tags, if present
-       # https://github.com/MailScanner/v5/issues/320
-       $squashedtext =~ s/(?:&lt;|&gt;)//g; 
-       my @list = split(/@/, $squashedtext);
-       $emailuser = $list[0];
-       $squashedtext = $list[1]; # Remove username of email addresses
+    if ( $DisarmLinkURL =~ m/^ma[il]+to[:;]/i ) {
```
Comment (Skywalker-11) [inline common/usr/share/MailScanner/perl/MailScanner/Message.pm]: What use case does this have [il]+ ? malito:??

File: common/usr/share/MailScanner/perl/MailScanner/Message.pm (hunk 1 above)
Comment (shawniverson) [reply common/usr/share/MailScanner/perl/MailScanner/Message.pm]: @Skywalker-11 This is Julian's original regex for the mailto, my best guess is to catch some accidental or deliberate typos.

Comment (shawniverson) [reply common/usr/share/MailScanner/perl/MailScanner/Message.pm]: I thought about this some more and I think I'm going to go ahead and remove the funny regex in this context of use.  'mailto:' with any other deviation is not really a 'mailto:' so it therefore does not need to be disarmed unless I'm missing something...

Comment (msapiro) [reply common/usr/share/MailScanner/perl/MailScanner/Message.pm]: I have no idea what the original reason for this was, but I agree that if the href doesn't match /mailto:/i it won't be seen as a mailto: by a browser or mua and doesn't need disarming.

Comment (Skywalker-11) [reply common/usr/share/MailScanner/perl/MailScanner/Message.pm]: There would also be no program registered for that URI so it won't do anything at all but throwing an error that the OS doesn't know any program to open it with

Comment (shawniverson) [reply common/usr/share/MailScanner/perl/MailScanner/Message.pm]: https://github.com/MailScanner/v5/pull/339

File: common/usr/share/MailScanner/perl/MailScanner/Message.pm (hunk 2 above)
Comment (shawniverson) [reply common/usr/share/MailScanner/perl/MailScanner/Message.pm]: https://github.com/MailScanner/v5/pull/339

File: common/usr/share/MailScanner/perl/MailScanner/Message.pm (hunk 3 above)
Comment (shawniverson) [reply common/usr/share/MailScanner/perl/MailScanner/Message.pm]: https://github.com/MailScanner/v5/pull/339

**Pull request merged**